### PR TITLE
AEA-1348 remove dummy packages patch from the aea loader

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

remove dummy packages patch from the aea loader

## Fixes

**https://github.com/fetchai/agents-aea/issues/1983**
## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/components/base.py [hunk 1]
```diff
@@ -131,9 +130,6 @@ def load_aea_package(configuration: ComponentConfiguration) -> None:
     prefix_author = prefix_root + f".{configuration.author}"
     prefix_pkg_type = prefix_author + f".{configuration.component_type.to_plural()}"
     prefix_pkg = prefix_pkg_type + f".{configuration.name}"
-    sys.modules[prefix_root] = types.ModuleType(prefix_root)
-    sys.modules[prefix_author] = types.ModuleType(prefix_author)
-    sys.modules[prefix_pkg_type] = types.ModuleType(prefix_pkg_type)
```
Comment (DavidMinarsch) [inline aea/components/base.py:136]: @marcofavorito why was this added in the first place?

Comment (marcofavorito) [reply aea/components/base.py:136]: to allow imports for prefixes of import paths. And to improve the error message in case of wrong prefix path. (e.g. existing author but missing package)

File: packages/fetchai/skills/ml_train/tasks.py [hunk 2]
```diff
@@ -71,10 +68,11 @@ def _make_model() -> Any:
     def execute(self, *args: Any, **kwargs: Any) -> Any:
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/ml_train/tasks.py:68]: return type could be specified more accurately

File: tests/conftest.py [hunk 3]
```diff
@@ -367,6 +363,20 @@
 ]
 
 
+@contextmanager
```
Comment (DavidMinarsch) [inline tests/conftest.py:366]: Is this used automatically for each test?

File: aea/components/base.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/components/base.py:136]: Where do we need `imports for prefixes of import paths.`?

Re error message, we then need to improve the error logging there to ensure this doesn't get worse as a result of this PR.

File: tests/conftest.py (hunk 3 above)
Comment (solarw) [reply tests/conftest.py:366]: no, need to set explicitly for every test.
probably will apply later to specific tests.

just want to complete this pr

File: aea/components/base.py (hunk 1 above)
Comment (solarw) [reply aea/components/base.py:136]: done

**Review by 5A11 — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: aea/skills/tasks.py [hunk 4]
```diff
@@ -64,7 +63,7 @@ def __call__(self, *args: Any, **kwargs: Any) -> "Task":
         finally:
             self._is_executed = True
             self.teardown()
-        return self
```
Comment (DavidMinarsch) [inline aea/skills/tasks.py:67]: This breaks what we have in the gym skill. Why not return self? This let's you check task status etc.

File: tests/test_task.py [hunk 5]
```diff
@@ -107,25 +107,8 @@ def test_task_manager_task_object(self):
         task_result = self.task_manager.get_task_result(task_id)
         assert isinstance(task_result, AsyncResult)
 
-        expected_task = task_result.get(self.WAIT_TIMEOUT)
```
Comment (DavidMinarsch) [inline tests/test_task.py:110]: See above!
